lib/genesis/helpers.go

```diff
@@ -157,23 +159,126 @@ type keyValue struct {
 	iVal     []interface{}
 }
 
+func generatePalletKeyValue(k string, v map[string]interface{}, res map[string]string) (bool, error) {
+	jsonBody, err := json.Marshal(v)
+	if err != nil {
+		return false, err
+	}
+
+	switch k {
+	case "Society":
+		s := &society{}
+		if err = json.Unmarshal(jsonBody, s); err != nil {
+			return false, err
+		}
+
+		err = generateKeyValue(s, k, res)
+		if err != nil {
+			return false, err
+		}
+	case "Staking":
+		st := &staking{}
+		if err = json.Unmarshal(jsonBody, st); err != nil {
+			return false, err
+		}
+
+		err = generateKeyValue(st, k, res)
+		if err != nil {
+			return false, err
+		}
+	case "Contracts":
+		c := &contracts{}
+		if err = json.Unmarshal(jsonBody, c); err != nil {
+			return false, err
+		}
+
+		err = generateContractKeyValue(c, k, res)
+		if err != nil {
+			return false, err
+		}
+	case "Session":
+		s := &session{}
+		if err = json.Unmarshal(jsonBody, s); err != nil {
+			return false, err
+		}
+
+		err = generateSessionKeyValue(s, k, res)
+		if err != nil {
+			return false, err
+		}
+	case "Instance1Collective":
+		ic := &instance1Collective{}
+		if err = json.Unmarshal(jsonBody, ic); err != nil {
+			return false, err
+		}
+
+		err = generateKeyValue(ic, k, res)
+		if err != nil {
+			return false, err
+		}
+	case "Instance2Collective":
+		ic := &instance2Collective{}
+		if err = json.Unmarshal(jsonBody, ic); err != nil {
+			return false, err
+		}
+
+		err = generateKeyValue(ic, k, res)
+		if err != nil {
+			return false, err
+		}
+	case "Instance1Membership":
+		im := &instance1Membership{}
+		if err = json.Unmarshal(jsonBody, im); err != nil {
+			return false, err
+		}
+
+		err = generateKeyValue(im, k, res)
+		if err != nil {
+			return false, err
+		}
+	case "PhragmenElection":
+		pe := &phragmenElection{}
+		if err = json.Unmarshal(jsonBody, pe); err != nil {
+			return false, err
+		}
+
+		err = generateKeyValue(pe, k, res)
+		if err != nil {
+			return false, err
+		}
```

Comment:
this code seems to be the same for every case, is it possible to move it out of the switch case?

Reply:
Done